### PR TITLE
fixes: fix topology hint tests, hint generation for non-empty sysroot.

### PR DESCRIPTION
**Pull request by klihub**
This PR fixes topology hint units tests, test data, and hint generation for non-empty sysroots. In particular this PR has the following changes:
- clean sysroot upon setting
- return resolved device paths without sysroot prefix (as expected by NewTopologyHints)
- don't run topology tests parallel (tests set/clear sysroot which is a global shared setting)
- add missing test data for /dev/null

### CODE REVIEW

File: pkg/topology/topology.go [hunk 1]
```diff
@@ -300,6 +300,12 @@ func findSysFsDevice(devType string, major, minor int64) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("failed to get realpath for %s: %w", devPath, err)
 	}
+	if sysRoot != "" {
+		realDevPath = strings.TrimPrefix(realDevPath, sysRoot)
+		if realDevPath != "" && realDevPath[0] != '/' {
+			realDevPath = "/" + realDevPath
+		}
```
Comment (fmuyassarov) [inline pkg/topology/topology.go:307]: just curious, why do we need to add a slash ? to indicate it as path?

**Review by fmuyassarov — COMMENTED** [review]
@klihub this looks good to me. I have a quick question in line. Also, your https://github.com/containers/nri-plugins/pull/326/commits/2024ea90071d72717c81e03d3a544478ef06d754 commit is doing something else it seems? There is a tarball added which might be added accidentally ...?

**Comment by klihub** [comment]
> @klihub this looks good to me. I have a quick question in line. Also, your [2024ea9](https://github.com/containers/nri-plugins/commit/2024ea90071d72717c81e03d3a544478ef06d754) commit is doing something else it seems? There is a tarball added which might be added accidentally ...?

That's the 2nd bullet point above. It adds missing data for /dev/null. One of the test cases tries to resolve /dev/null to a sysfs path, but our sample sysfs test data lacks /dev/null information. That test case only worked by accident, due to another bug. Test cases were run in parallel, and each test case which sets up the package global non-standard sysfs path also clears/resets it on tetscase teardown. Since tests were run in parallel, the first and fastest test almost always had time to reset the sysfs path, and the /dev/null resolution test was resolving /dev/null using the hosts sysfs, instead of our sample test data. Once that bug was fixed, the /dev/null test started understandably failing. Adding the missing data for that test case then fixed this and it started working for real, not just by accident.

**Comment by fmuyassarov** [comment]
> @fmuyassarov So this PR consists of 3 separate fixes to the topology package in 3 separate commits. I tried to describe each with a one-liner bullet point in the PR description [above](https://github.com/containers/nri-plugins/pull/326#issue-2325921797).

Yes I know. I just didn't realize that the tarball was the test data.

**Review by fmuyassarov — APPROVED** [review]
LGTM

File: pkg/topology/topology.go [hunk 2]
```diff
@@ -300,6 +300,12 @@ func findSysFsDevice(devType string, major, minor int64) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("failed to get realpath for %s: %w", devPath, err)
 	}
+	if sysRoot != "" {
+		realDevPath = strings.TrimPrefix(realDevPath, sysRoot)
```
Comment (kad) [inline pkg/topology/topology.go:304]: should we better to check that if the `realDevPath` prefixed with `sysRoot`, instead of just checking if `sysRoot` is not empty?

Comment (kad) [reply pkg/topology/topology.go:304]: As a side note, in `SetSysRoot` we can use `filepath.Clean()` to make sure that the value is correct, so operations with those paths will be more predictable. e.g. in line 181.

**Comment by askervin** [comment]
Validated in real environment with multiple PCI devices. Works great. Thank you, @klihub. I think Draft status can be removed.

File: pkg/topology/topology.go (hunk 2 above)
Comment (klihub) [reply pkg/topology/topology.go:308]: Updated to check prefix against cleaned sysroot.

Comment (askervin) [reply pkg/topology/topology.go:308]: Path returned by `filepath.Clean()` is promised not to end with "/" except for in the case of the root directory. Therefore, if user calls `SetSysRoot("/")`, the latest change seems to make it possible that `findSysFsDevice()` returns paths that do not start with "/". I'm not sure if this will cause issues, and whether or not it is possible to call `SetSysRoot("/")`, but at least it is more possible than it was before.

**Review by askervin — COMMENTED** [review]
A question on "always absolute path" promise from findSysFsDevice().

File: pkg/topology/topology.go (hunk 2 above)
Comment (klihub) [reply pkg/topology/topology.go:318]: Updated to 1) only use implicit "/" as sysroot and 2) always resolve relative sysroot to absolute one, panicking if this fails. Added a few basic test cases for `SetSysRoot()`.

**Review by askervin — APPROVED** [review]
Thanks @klihub, LGTM.
